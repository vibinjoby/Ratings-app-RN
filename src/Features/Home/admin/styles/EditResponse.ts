import { StyleSheet } from 'react-native'

import Colors from '../../../../utilities/colors'
import Typography from '../../../../utilities/typography'

export default StyleSheet.create({
  container: {
    backgroundColor: Colors.white,
    borderRadius: 8,
    padding: 15,
  },
  title: {
    ...Typography.BoldFont,
    fontSize: 20,
    marginBottom: 5,
  },
  textInp: {
    backgroundColor: Colors.borderColor,
    padding: 10,
    marginBottom: 10,
  },
  stars: {
    marginBottom: 10,
  },
  btnWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginTop: 25,
  },
  cancelBtn: {
    minWidth: 115,
    borderRadius: 5,
    backgroundColor: Colors.borderColor,
  },
  yesBtn: {
    minWidth: 115,
    borderRadius: 5,
    backgroundColor: Colors.appOrange,
  },
})
