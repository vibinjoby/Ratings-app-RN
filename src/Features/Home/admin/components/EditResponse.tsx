import React, { useState } from 'react'
import { View, TextInput, Text } from 'react-native'
import Modal from 'react-native-modal' //@ts-ignore
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scrollview'

import Colors from '../../../../utilities/colors'
import Button from '../../../../components/AppButton'
import Stars from '../../../../components/AppStars'
import styles from '../styles/EditResponse'

export interface EditResponseProps {
  customerResp: string
  ownerResp?: string
  isVisible: boolean | undefined
  ratings: number
  onEdit: (comments: string, ownerReply: string, ratings: number) => void
  onDismiss: () => void
}

const EditResponse: React.FC<EditResponseProps> = ({
  customerResp,
  ownerResp,
  isVisible,
  onDismiss,
  ratings,
  onEdit,
}: EditResponseProps) => {
  const [customerResponse, setCustResponse] = useState(customerResp)
  const [ownerResponse, setOwnerResponse] = useState(ownerResp ?? '')
  const [selectedStars, setSelectedStars] = useState(ratings)

  const onSave = () => {
    onEdit(customerResponse, ownerResponse, selectedStars)
  }

  return (
    <KeyboardAwareScrollView>
      <Modal
        isVisible={isVisible}
        onBackdropPress={onDismiss}
        backdropOpacity={0.5}
        hasBackdrop
        useNativeDriver={false}
      >
        <View style={styles.container}>
          <Text style={styles.title}>Customer Review</Text>
          <TextInput
            testID="customerRespId"
            autoFocus
            blurOnSubmit
            placeholder="Enter Customer response"
            style={styles.textInp}
            value={customerResponse}
            onChangeText={(text) => setCustResponse(text)}
            multiline
          />
          <Text style={styles.title}>Ratings</Text>
          <View style={styles.stars}>
            <Stars
              testID="star"
              selectedStars={selectedStars}
              onSelection={(star) => setSelectedStars(star)}
              selectable
              selectedColor={Colors.appOrange}
            />
          </View>
          <Text style={styles.title}>Owner Response</Text>
          <TextInput
            testID="ownerRespId"
            placeholder="Enter Owner response"
            style={styles.textInp}
            value={ownerResponse}
            onChangeText={(text) => setOwnerResponse(text)}
            multiline
          />
          <View style={styles.btnWrapper}>
            <Button
              testID="cancel"
              title={'Cancel'}
              onPress={onDismiss}
              customStyle={styles.cancelBtn}
              textColor={Colors.black}
            />
            <Button
              testID="save"
              title={'Save'}
              disabled={!customerResponse || !selectedStars}
              onPress={onSave}
              customStyle={styles.yesBtn}
            />
          </View>
        </View>
      </Modal>
    </KeyboardAwareScrollView>
  )
}

export default EditResponse
