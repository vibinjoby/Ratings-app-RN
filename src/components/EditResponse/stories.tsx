import React, { useState } from 'react'
import { StyleSheet, View } from 'react-native'
import { storiesOf } from '@storybook/react-native'

import EditResponse from './component'
import Button from '../Button'

const Component = () => {
  const [isVisible, setIsVisible] = useState(false)

  const toggleModal = () => {
    setIsVisible((val) => !val)
  }

  return (
    <View style={styles.container}>
      <Button title="Show modal" onPress={toggleModal} />
      <EditResponse
        isVisible={isVisible}
        customerResp="Lorem ipsium"
        ownerResp="Thanks for submitting your response but we dont value your time nor have interest to give you adequate feedback"
        onDismiss={toggleModal}
        onEdit={toggleModal}
        ratings={4}
      />
    </View>
  )
}

storiesOf('Components', module).add('EditResponse', () => <Component />)

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
})
