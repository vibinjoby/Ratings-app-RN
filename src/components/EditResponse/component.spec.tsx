import 'react-native'
import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

import EditResponse from './component'

describe('EditResponse', () => {
  const mockFn = jest.fn()
  const MOCK_CUSTOMER_RESPONSE = 'thanks for the food'
  const MOCK_OWNER_RESPONSE = 'Thanks for the review'
  const MOCK_RATINGS = 4
  const mockDismiss = jest.fn()
  const mockSave = jest.fn()

  it('should render', () => {
    const { toJSON } = render(
      <EditResponse
        isVisible={true}
        customerResp="Lorem ipsium"
        ownerResp="Thanks for submitting your response but we dont value your time nor have interest to give you adequate feedback"
        onDismiss={mockFn}
        onEdit={mockFn}
        ratings={4}
      />,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should save with all inputs', () => {
    const { getByTestId } = render(
      <EditResponse
        isVisible={true}
        customerResp={''}
        ownerResp={''}
        onDismiss={mockDismiss}
        onEdit={mockSave}
        ratings={MOCK_RATINGS}
      />,
    )
    fireEvent.changeText(getByTestId('customerRespId'), MOCK_CUSTOMER_RESPONSE)
    fireEvent.changeText(getByTestId('ownerRespId'), MOCK_OWNER_RESPONSE)
    fireEvent.press(getByTestId('save'))
    expect(mockSave).toBeCalled()
  })
})
