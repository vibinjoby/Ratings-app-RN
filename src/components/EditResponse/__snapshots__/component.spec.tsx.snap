// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditResponse should render 1`] = `
<RCTScrollView
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  getTextInputRefs={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onScroll={[Function]}
  scrollEventThrottle={200}
  scrollToBottomOnKBShow={false}
  scrollToInputAdditionalOffset={75}
  startScrolledToBottom={false}
>
  <View>
    <Modal
      animationType="none"
      deviceHeight={null}
      deviceWidth={null}
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[MockFunction]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onRequestClose={[Function]}
      panResponderThreshold={4}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      statusBarTranslucent={false}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={true}
    >
      <View
        accessible={true}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        deviceHeight={null}
        deviceWidth={null}
        forwardedRef={[Function]}
        hideModalContentWhileAnimating={false}
        onBackdropPress={[MockFunction]}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        panResponderThreshold={4}
        pointerEvents="box-none"
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        statusBarTranslucent={false}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 8,
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Poppins-Bold",
                "fontSize": 20,
                "marginBottom": 5,
              }
            }
          >
            Customer Review
          </Text>
          <TextInput
            autoFocus={true}
            blurOnSubmit={true}
            multiline={true}
            onChangeText={[Function]}
            placeholder="Enter Customer response"
            style={
              Object {
                "backgroundColor": "#E2E2E2",
                "marginBottom": 10,
                "padding": 10,
              }
            }
            testID="customerRespId"
            value="Lorem ipsium"
          />
          <Text
            style={
              Object {
                "fontFamily": "Poppins-Bold",
                "fontSize": 20,
                "marginBottom": 5,
              }
            }
          >
            Ratings
          </Text>
          <View
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <RCTScrollView
              ItemSeparatorComponent={[Function]}
              data={
                Array [
                  1,
                  2,
                  3,
                  4,
                  5,
                ]
              }
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={true}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEnabled={false}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              viewabilityConfigCallbackPairs={Array []}
            >
              <View>
                <View
                  onLayout={[Function]}
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                      },
                      null,
                    ]
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    nativeID="animatedComponent"
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="star0"
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#FF5823",
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="font-icon 0"
                    >
                      
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "marginRight": 5,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
                <View
                  onLayout={[Function]}
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                      },
                      null,
                    ]
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    nativeID="animatedComponent"
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="star1"
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#FF5823",
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="font-icon 1"
                    >
                      
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "marginRight": 5,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
                <View
                  onLayout={[Function]}
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                      },
                      null,
                    ]
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    nativeID="animatedComponent"
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="star2"
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#FF5823",
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="font-icon 2"
                    >
                      
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "marginRight": 5,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
                <View
                  onLayout={[Function]}
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                      },
                      null,
                    ]
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    nativeID="animatedComponent"
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="star3"
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#FF5823",
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="font-icon 3"
                    >
                      
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "marginRight": 5,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
                <View
                  onLayout={[Function]}
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                      },
                      null,
                    ]
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    nativeID="animatedComponent"
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="star4"
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#A5A6B7",
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="font-icon 4"
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
          <Text
            style={
              Object {
                "fontFamily": "Poppins-Bold",
                "fontSize": 20,
                "marginBottom": 5,
              }
            }
          >
            Owner Response
          </Text>
          <TextInput
            multiline={true}
            onChangeText={[Function]}
            placeholder="Enter Owner response"
            style={
              Object {
                "backgroundColor": "#E2E2E2",
                "marginBottom": 10,
                "padding": 10,
              }
            }
            testID="ownerRespId"
            value="Thanks for submitting your response but we dont value your time nor have interest to give you adequate feedback"
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-around",
                "marginTop": 25,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#E2E2E2",
                  "borderRadius": 5,
                  "minWidth": 115,
                  "opacity": 1,
                  "padding": 14,
                }
              }
              testID="cancel"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Poppins-Regular",
                      "fontSize": 18,
                    },
                    Object {
                      "color": "#181725",
                    },
                  ]
                }
              >
                Cancel
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FF5823",
                  "borderRadius": 5,
                  "minWidth": 115,
                  "opacity": 1,
                  "padding": 14,
                }
              }
              testID="save"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Poppins-Regular",
                      "fontSize": 18,
                    },
                    Object {},
                  ]
                }
              >
                Save
              </Text>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</RCTScrollView>
`;
